### PR TITLE
added template support for web mojo.

### PR DESCRIPTION
**Pull request by francesco-kriegel**
The current version does not support user-defined templates for HTML files when creating a WebStart bundle. This branch supports this feature.

###### 1. Add HTML templates to `src/main/deploy/templates`.

A sample template for embedded launch:
```
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
		<title>Your JavaFX Application</title>
		#DT.SCRIPT.CODE#
		#DT.EMBED.CODE.ONLOAD#
		<style>
			html, body, #javafx-app-placeholder, #fxApp {
				margin: 0;
				overflow: hidden;
				padding: 0;
				width: 100%;
				height: 100%;
			}
		</style>
	</head>
	<body>
		<div id='javafx-app-placeholder'></div>
	</body>
</html>
```
A sample template for windowed launch:
```
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
		<title>Your JavaFX Application</title>
		#DT.SCRIPT.CODE#
	</head>
	<body>
		<!-- 		<a href='index.html' onclick="return #DT.LAUNCH.CODE#">Launch Your JavaFX Application in its own window</a> -->
		<script>
			#DT.LAUNCH.CODE#
			function reload() {
				window.location.href = "index.html";
			}
			window.setTimeout(reload, 6000);
		</script>
	</body>
</html>
```
###### 2. Reference them in the plugin configuration as follows:
```
<plugin>
	<groupId>com.zenjava</groupId>
	<artifactId>javafx-maven-plugin</artifactId>
	<version>8.1.3-SNAPSHOT</version>
	<configuration>
		<templates>
			<template>template1.html</template>
			<template>template2.html</template>
		</templates>
	<configuration>
</plugin>
```
###### 3. Run `mvn jfx:web` or add the web mojo to your build lifecycle.

### CODE REVIEW

File: src/main/java/com/zenjava/javafx/maven/plugin/WebMojo.java [hunk 1]
```diff
@@ -198,6 +205,16 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             String embeddedWidth = this.embeddedWidth != null ? this.embeddedWidth : String.valueOf(width);
             String embeddedHeight = this.embeddedHeight != null ? this.embeddedHeight : String.valueOf(height);
             deployParams.setEmbeddedDimensions(embeddedWidth, embeddedHeight);
+            
+            if (templates != null)
+                for (String template : templates) {
+                  final File in = new File(project.getBasedir(), "src/main/deploy/templates/" + template);
```
Comment (maxd) [inline src/main/java/com/zenjava/javafx/maven/plugin/WebMojo.java]: Will be better to use OS specific path separators i.e. use [Path#get](http://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get%28java.lang.String,%20java.lang.String...%29).

File: src/main/java/com/zenjava/javafx/maven/plugin/WebMojo.java [hunk 2]
```diff
@@ -163,6 +165,11 @@
      */
     protected String keyStoreType;
 
+    /**
+     * @parameter
```
Comment (maxd) [inline src/main/java/com/zenjava/javafx/maven/plugin/WebMojo.java]: Required description of this parameter. Seems like you already done it PR description. Just need to convert it to JavaDoc.

**Comment by FibreFoX** [comment]
Hi @francesco-kriegel , thanks for this addition. Could you please add some description to mojo-parameter ```templates``` and change templates to ```List<File>``` ? This avoids for OS-specific paths-stuff and makes template-position open to user-choice.

**Comment by francesco-kriegel** [comment]
Hi @FibreFoX , thanks for your reply. I included the requested change. Furthermore, I finally pushed my other changes I made some weeks ago. The lifecycle bindings were removed to prevent multiple executions of the jar plugin. Maybe these issues have already been resolved with the newly introduced lifecycle bindings. There are some further parameters for adjusting the native and web deployment. Possibly some of the additional parameters in the WebMojo can be integrated in the NativeMojo, too. I have not tested this yet. In the JarMojo an option for using an existing jar, e.g., a jar-with-dependencies from the maven-assembly-plugin, was added. This avoids problems when signing third-party libraries that occur sometimes.

If you have problems merging the pull request, then let me know. I may also open a new pull request based on the latest version in your respository, if neccessary.

**Comment by FibreFoX** [comment]
Hi @francesco-kriegel, would be nice if you rebase your repository to current master (instead of just pull). JarMojo already got lifecycle-bindings removed (I introduced CliJarMojo).

**Comment by francesco-kriegel** [comment]
Hi @FibreFoX , now the code is rebased to your latest version. My local changes have been merged and pushed as you can see above. Maybe you need to reformat the code since I am using a different code formatter style.

**Comment by maxd** [comment]
@francesco-kriegel We won't merge this PR because it totally broke current interface of plugin (i.e. `updateExistingJar` field renamed to `existingJar`). 

+ Seems like this PR trays to fix several problems. Will be better to have separate PR for them.

So, I close it. Please provide separate PR with `template support` functionality only and current interface.

**Comment by FibreFoX** [comment]
Just for information: some changes here are covered with PR #117

**Pull request closed without merging**